src/html5/parser/helper.rs

```diff
@@ -239,11 +240,16 @@ impl Html5Parser<'_> {
     }
 
     pub fn insert_text_element(&mut self, token: &Token) {
-        let node = self.create_node(token, HTML_NAMESPACE);
-        let node_id = self.document.get_mut().add_new_node(node);
+        // Skip empty text nodes
+        if let Token::Text(text) = token {
```

Comment:
It seems possible that empty tokens arrive here. We have to make sure we don't add them to the tree